File: docs/src/test/java/org/springframework/security/docs/servlet/oauth2/resourceserver/methodsecurityhasscope/MessageService.java
package org.springframework.security.docs.servlet.oauth2.resourceserver.methodsecurityhasscope;


import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.stereotype.Service;

@Service
class MessageService {

	// tag::protected-method[]
	@PreAuthorize("@oauth2.hasScope('message:read')")
	String readMessage() {
		return "message";
	}
	// end::protected-method[]
}

File: docs/src/test/kotlin/org/springframework/security/kt/docs/servlet/oauth2/resourceserver/methodsecurityhasscope/MessageService.kt
package org.springframework.security.kt.docs.servlet.oauth2.resourceserver.methodsecurityhasscope

import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.stereotype.Service


@Service
open class MessageService {
    // tag::protected-method[]
    @PreAuthorize("@oauth2.hasScope('message:read')")
    open fun readMessage(): String {
        return "message"
    }
    // end::protected-method[]
}
